src/core/include/openvino/op/grid_sample.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/attribute_adapter.hpp"
+#include "openvino/op/op.hpp"
+#include "openvino/op/util/attr_types.hpp"
+
+namespace ov {
+namespace op {
+namespace v9 {
+
+/// \brief Operator performing interpolated sampling of the input tensor
+/// \ingroup ov_ops_cpp_api
+class OPENVINO_API GridSample : public Op {
+public:
+    OPENVINO_OP("GridSample", "opset9");
+    BWDCMP_RTTI_DECLARATION;
+
+    enum class InterpolationMode { BILINEAR, BICUBIC, NEAREST };
+    enum class PaddingMode { ZEROS, BORDER, REFLECTION };
+
+    /// \brief A Structure which contains all GridSample attributes
+    struct Attributes {
+        // A flag which specifies whether to align the grid extrema values with the borders or center points
+        // of the input tensor's border pixels.
+        bool align_corners = false;
+        // Specifies the type of interpolation: `bilinear`, `bicubic`, `nearest`
+        InterpolationMode mode = InterpolationMode::BILINEAR;
+        // Specifies how the out-of-bounds coordinates should be handled: `zeros`, `border`, `reflection`
+        PaddingMode padding_mode = PaddingMode::ZEROS;
+    };
+
+    GridSample() = default;
+    /// \brief Constructs a GridSample operation
+    ///
+    /// \param data   Input data tensor (input image)
+    /// \param grid   Normalized interpolation coordinates
+    /// \param attrs  GridSample attributes
+    GridSample(const Output<Node>& data, const Output<Node>& grid, const Attributes& attributes);
```

Comment:
I am puzzled with this constructor. I know that there are two approaches to it, one using attributes (like here), second one is just stating attributes one by one in constructor... Different ops use different approaches. We should follow it up offline and decide on one "true" approach (or implement even both of them) as this current state, in my opinion, leads to APIs inconsistencies.